### PR TITLE
Topic/create field map while adding trials.

### PR DESCRIPTION
**Pull request by aco46**
I have added in the add trial dialog options to generate field map on the fly while adding a trial.
It has two use cases:
 **Use Case 1:** when plot_layout_format is specified alone, by default rep/block number would be used as the row_number and col_number given the trial design selected.
**Use Case 2:** You can select a plot _layout_format and enter a row_number. Base on the number of plots, column_number will be generated.

### CODE REVIEW

File: lib/CXGN/Trial/TrialDesign.pm [hunk 1]
```diff
@@ -463,13 +633,54 @@ sub _get_alpha_lattice_design {
...
+    }
+
   for (my $i = 0; $i < scalar(@converted_plot_numbers); $i++) {
     my %plot_info;
     $plot_info{'stock_name'} = $stock_names[$i];
     $plot_info{'block_number'} = $block_numbers[$i];
     $plot_info{'plot_name'} = $converted_plot_numbers[$i];
     $plot_info{'rep_number'} = $rep_numbers[$i];
     $plot_info{'is_a_control'} = exists($control_names_lookup{$stock_names[$i]});
+    if ($fieldmap_row_numbers[$i]){
```
Comment (nickmorales) [inline lib/CXGN/Trial/TrialDesign.pm]: for alpha design check page 20 for enumeration of plots. https://cran.r-project.org/web/packages/agricolae/vignettes/tutorial.pdf

File: lib/CXGN/Trial/TrialDesign.pm [hunk 2]
```diff
@@ -343,13 +441,55 @@ sub _get_rcbd_design {
...
+
   for (my $i = 0; $i < scalar(@converted_plot_numbers); $i++) {
     my %plot_info;
     $plot_info{'stock_name'} = $stock_names[$i];
     $plot_info{'block_number'} = $block_numbers[$i];
     $plot_info{'plot_name'} = $converted_plot_numbers[$i];
     $plot_info{'rep_number'} = 1;
     $plot_info{'is_a_control'} = exists($control_names_lookup{$stock_names[$i]});
+    if ($fieldmap_row_numbers[$i]){
+      $plot_info{'row_number'} = $fieldmap_row_numbers[$i];
```
Comment (nickmorales) [inline lib/CXGN/Trial/TrialDesign.pm]: for RCDB check page 11 for enumeration of plots https://cran.r-project.org/web/packages/agricolae/vignettes/tutorial.pdf

File: js/CXGN/BreedersToolbox/AddTrial.js [hunk 3]
```diff
@@ -459,6 +467,15 @@ jQuery(document).ready(function ($) {
 	}
     });
 
+    $("#show_field_map_options").click(function () {
```
Comment (aco46) [inline js/CXGN/BreedersToolbox/AddTrial.js]: Thanks for the observation. Would look odd to show for all designs.

File: lib/CXGN/Trial/TrialDesign.pm [hunk 4]
```diff
@@ -312,6 +390,26 @@ sub _get_rcbd_design {
   } else {
     die "Number of blocks not specified\n";
   }
+  sub isint{
```
Comment (aco46) [inline lib/CXGN/Trial/TrialDesign.pm]: I would have to move it out of each of the design function and have it as a separate function.

**Comment by aco46** [comment]
Hey Nick, would you want to take a second look before I finally merge the branch to master. I've improved the test for Alpha design.

**Pull request merged**